sdk/python/feast/client.py

```diff
@@ -339,13 +341,17 @@ def archive_project(self, project):
         """
 
         self._connect_core()
-        self._core_service_stub.ArchiveProject(
-            ArchiveProjectRequest(name=project),
-            timeout=self._config.getint(CONFIG_GRPC_CONNECTION_TIMEOUT_DEFAULT_KEY),
-        )  # type: ArchiveProjectResponse
+        try:
+            self._core_service_stub.ArchiveProject(
+                ArchiveProjectRequest(name=project),
+                timeout=self._config.getint(CONFIG_GRPC_CONNECTION_TIMEOUT_DEFAULT_KEY),
+            )  # type: ArchiveProjectResponse
+        except grpc.RpcError as e:
+            raise grpc.RpcError(e.details())
 
+        # revert to the default project
         if self._project == project:
-            self._project = ""
+            self._project = FEAST_DEFAULT_OPTIONS[CONFIG_PROJECT_KEY]
```

Comment:
I think its cleaner to reset the project instead of setting it. Otherwise we don't know if the active config is the default or not.

Reply:
When the user first runs, post `pip install feast`:
`
from feast import Client
client = Client()
`
It imports default options `FEAST_DEFAULT_OPTIONS` which sets the project to `default`.
This change makes it consistent that client will revert to `default` when project is otherwise unspecified or an existing one archived.